### DISCUSSION TITLE
Create new folder daily and write files

### DISCUSSION THREAD

**Discussion by wcotis** [nextgenhealthcare/connect#6321] [Q&A] [answered]
Looking to use Mirth Connect to receive files and write them to a backup NAS. Need to create a new folder each day and write that days files then the next day and so on.
Can this be done in Mirth Connect? How?

**Comment by kirbykn2** [comment]
Yes it can be done.   Plenty of information in the docs and forums to get
started.  Do you have an existing Mirth environment?

https://docs.nextgen.com/bundle/Mirth_User_Guide_40/page/connect/connect/topics/c_Using_Open_Source_Software_mirth_connect_ug.html
https://github.com/nextgenhealthcare/connect/issues/5320


On Fri, Oct 11, 2024 at 2:22 PM wcotis ***@***.***> wrote:

> Looking to use Mirth Connect to receive files and write them to a backup
> NAS. Need to create a new folder each day and write that days files then
> the next day and so on.
> Can this be done in Mirth Connect? How?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/6321>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWD7SRSBYOY3PYK3GVQDZ3AJOHAVCNFSM6AAAAABPZNNPNGVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGMYDMOBYG4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Reply by wcotis** [reply]
Yes, Mirth Connect 4.1.0
I have the basics set up, DICOM send to the Mirth and it then uses "File Writer" to write the files to a specified folder. I need to create a new folder for each day and write to it.

**Reply by pacmano1** [reply, marked as answer]
Mirth will create the named folder you specify in the folder settings.  

![image](https://github.com/user-attachments/assets/ef0f738c-3ca2-4cd0-b36c-204c753e506d)

**Comment by wcotis** [comment]
Thanks, I was able to get this sorted out yesterday, used the "${date.get('yyyy-M-d')}" mapping and it is working.

**Reply by wcotis** [reply]
The "Directory" statement being used, "C:/${date.get('yyyy-M-d')}/${vPID}" is getting me close to what is needed.

This places each folder and file for the date into one folder with that date - folder "2025-1-15" created and multiple subfolders are written there as the folder exists after the first files are received.

I need to have these "daily" folders written to a "monthly" folder.  "2025-1" with the daily folders as shown above written in this folder.

How can these statement be nested in Mirth in the "Directory" location? I have searched but find nothing about nesting or concatenating these statements.

**Reply by pacmano1** [reply]
```
/tmp/${date.get('yyyy-M')}/${date.get('yyyy-M-d')}/${vPID}
```
On linux of course which would result in:
```
SENT: File successfully written: /tmp/2025-1/2025-1-15/400960121/3.hl7
```

I would use two digits for day and month always though.

**Reply by wcotis** [reply]
Thank you.
